packages/loader/container-loader/src/layerCompatState.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	checkLayerCompatibility,
+	type ILayerCompatDetails,
+	type ILayerCompatSupportRequirements,
+} from "@fluid-internal/client-utils";
+import type { ICriticalContainerError } from "@fluidframework/container-definitions";
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+
+import { pkgVersion } from "./packageVersion.js";
+
+/**
+ * Loader's compatibility details that is exposed to the Runtime layer.
+ */
+export const LoaderCompatDetails: ILayerCompatDetails = {
+	/**
+	 * The package version of the Loader layer.
+	 */
+	pkgVersion,
+	/**
+	 * The current generation of the Loader layer.
+	 */
+	generation: 1,
+	/**
+	 * The features supported by the Loader layer across the Loader / Runtime boundary.
+	 */
+	supportedFeatures: new Set<string>(),
```

Comment:
Shouldn't this include `referenceSequenceNumbers`?

Reply:
I removed this because Container runtime treats the presence of `ILayerCompatDetails` on the container context as `referenceSequenceNumbers` is supported. If `ILayerCompatDetails` doesn't exist, it falls back to the old behavior.
The behavior change here is that new Loaders won't be able to pass `referenceSequenceNumbers: false` which isn't the intention since this is always true.

Reply:
Still, it feels like a special case that we don't need.